WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt

```diff
@@ -0,0 +1,77 @@
+package org.wordpress.android.ui.bloggingprompts
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.DiffUtil.Callback
+import androidx.recyclerview.widget.RecyclerView
+import org.wordpress.android.R
+import org.wordpress.android.databinding.BloggingPromptsListItemBinding
+import org.wordpress.android.util.extensions.setVisible
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class BloggingPromptsListAdapter() : RecyclerView.Adapter<BloggingPromptsListItemViewHolder>() {
+    private var items = listOf<BloggingPromptsListItem>()
+
+    fun update(newItems: List<BloggingPromptsListItem>) {
+        val diffResult = DiffUtil.calculateDiff(
+                BloggingPromptsListDiffCallback(
+                        items,
+                        newItems
+                )
+        )
+        items = newItems
+        diffResult.dispatchUpdatesTo(this)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BloggingPromptsListItemViewHolder {
+        val layoutInflater = LayoutInflater.from(parent.context)
+        val binding = BloggingPromptsListItemBinding.inflate(layoutInflater, parent, false)
+        return BloggingPromptsListItemViewHolder(binding)
+    }
+
+    override fun getItemCount() = items.size
+
+    override fun onBindViewHolder(holder: BloggingPromptsListItemViewHolder, position: Int) {
+        holder.bind(items[position])
+    }
+}
+
+class BloggingPromptsListItemViewHolder(
+    private val binding: BloggingPromptsListItemBinding
+) : RecyclerView.ViewHolder(binding.root) {
+    fun bind(prompt: BloggingPromptsListItem) {
+        with(binding) {
+            promptTitle.text = prompt.title
+            promptSubtitleAnswerCount.text =
+                    when (prompt.respondentsCount) {
+                        0 -> binding.root.context.getString(R.string.blogging_prompts_list_item_count_answers_zero)
+                        1 -> binding.root.context.getString(R.string.blogging_prompts_list_item_count_answers_one)
+                        else -> binding.root.context.getString(
+                                R.string.blogging_prompts_list_item_count_answers_many,
+                                prompt.respondentsCount
+                        )
+                    }
+            promptSubtitleDate.text =
+                    SimpleDateFormat("MMM d", Locale.getDefault()).format(prompt.date)
```

Comment:
Would it make sense to extract this mapping to another class (e.g. ViewModel) instead so that it can be unit tested? In that case maybe we could use `Locale` from `LocaleManagerWrapper`.

Also, would it make sense to extract the date format to a constant?

Reply:
Yes. Will do.

Reply:
This is done.